simplified-ui-catalog/src/main/java/org/nypl/simplified/ui/catalog/saml20/CatalogSAML20Fragment.kt

```diff
@@ -0,0 +1,229 @@
+      val fragment = CatalogSAML20Fragment()
+      fragment.arguments = arguments
+      return fragment
+    }
+  }
+
+  private lateinit var profiles: ProfilesControllerType
+  private lateinit var booksController: BooksControllerType
+  private lateinit var bookRegistry: BookRegistryType
+  private lateinit var buildConfig: BuildConfigurationServiceType
+  private lateinit var parameters: CatalogSAML20FragmentParameters
+  private lateinit var progress: ProgressBar
+  private lateinit var uiThread: UIThreadServiceType
+  private lateinit var viewModel: CatalogSAML20ViewModel
+  private lateinit var webView: WebView
+  private lateinit var account: AccountType
+  private val eventSubscriptions = CompositeDisposable()
+
+  override fun onCreate(savedInstanceState: Bundle?) {
+    super.onCreate(savedInstanceState)
+
+    this.parameters = this.requireArguments()[PARAMETERS_ID] as CatalogSAML20FragmentParameters
+
+    val services = Services.serviceDirectory()
+
+    this.booksController =
+      services.requireService(BooksControllerType::class.java)
+    this.bookRegistry =
+      services.requireService(BookRegistryType::class.java)
+    this.uiThread =
+      services.requireService(UIThreadServiceType::class.java)
+    this.buildConfig =
+      services.requireService(BuildConfigurationServiceType::class.java)
+    this.profiles =
+      services.requireService(ProfilesControllerType::class.java)
+
+    val book = this.bookRegistry.bookOrException(this.parameters.bookID)
+    val accountID = book.book.account
+
+    this.account = this.profiles.profileCurrent().account(accountID)
+  }
+
+  override fun onCreateView(
+    inflater: LayoutInflater,
+    container: ViewGroup?,
+    savedInstanceState: Bundle?
+  ): View? {
+    val layout = inflater.inflate(R.layout.book_saml20, container, false)
+    this.webView = layout.findViewById(R.id.saml20WebView)
+    this.progress = layout.findViewById(R.id.saml20progressBar)
+    return layout
+  }
+
+  override fun onStart() {
+    super.onStart()
+
+    this.viewModel =
+      ViewModelProviders.of(
+        this,
+        CatalogSAML20ViewModelFactory(
+          booksController = this.booksController,
+          bookRegistry = this.bookRegistry,
+          account = this.account,
+          bookID = this.parameters.bookID,
+          webViewDataDir = this.requireContext().getDir("webview", Context.MODE_PRIVATE)
+        )
+      ).get(CatalogSAML20ViewModel::class.java)
+
+    this.eventSubscriptions.add(
+      this.viewModel.events.subscribe(
+        this::onSAMLEvent,
+        this::onSAMLEventException,
+        this::onSAMLEventFinished
+      )
+    )
+
+    this.webView.webChromeClient = CatalogSAML20ChromeClient(this.progress)
+    this.webView.webViewClient = this.viewModel.webViewClient
+    this.webView.settings.javaScriptEnabled = true
+    this.webView.setDownloadListener(this::onDownloadStart)
+
+    if (this.viewModel.isWebViewClientReady) {
+      this.loadLoginPage()
```

Comment:
See previous comment about orientation changes.

You might want to skip this initial load if you have a `savedInstanceState`.